### PR TITLE
Polish on the build

### PR DESCRIPTION
**Pull request by costinm**
- fix linter 
- fix the install/e2e tests
- add back the 'deb'
- general cleanup and simplifications.

Note that since we moved to single repo we no longer need to create one deb per git tree. I've moved
the scripts from proxy, using the envoy binary downloaded part of init and testing. The 'raw VM' install
process will now consist of a single .deb install instead of 3.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -36,10 +37,13 @@ jobs:
...
       - checkout
-      - run: curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.7.4/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
-      - run: curl -Lo minikube https://storage.googleapis.com/minikube/releases/v0.24.1/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
+      - attach_workspace:
+          at:  /go
       - run:
+          name: "Start minikube"
           command: |
+            curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.7.4/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
+            curl -Lo minikube https://storage.googleapis.com/minikube/releases/v0.22.3/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
```
Comment (rshriram) [inline .circleci/config.yml]: why do you need these?

Comment (costinm) [reply .circleci/config.yml]: I didn't push the image yet ( bad karma to apply the changes before it is reviewed ), once the PR
is approved and image pushed I will remove this.

File: .circleci/Makefile [hunk 2]
```diff
@@ -1,21 +1,20 @@
 # Builder version
-VERSION ?= go1.9-k8s1.7.4
-PROJECT ?= istio
-HUB ?=
+CI_VERSION ?= go1.9-k8s1.7.4
```
Comment (rshriram) [inline .circleci/Makefile]: change this, as you have changed the docker file

File: .circleci/config.yml [hunk 3]
```diff
@@ -253,11 +282,17 @@ jobs:
...
-            make localTestEnv go-test KUBECONFIG=/go/src/istio.io/istio/.circleci/config
+            mkdir -p /go/out/tests
+            go get github.com/jstemmer/go-junit-report
+            trap "go-junit-report </go/out/tests/go-test-report.out > /go/out/tests/go-test-report.xml" EXIT
+            make localTestEnv go-test KUBECONFIG=/go/src/istio.io/istio/.circleci/config T=-v | tee /go/out/tests/go-test-report.out
       - store_artifacts:
           path: /go/bin
+          path: /go/out
+      - store_test_results:
+          path: /go/out/tests
```
Comment (rshriram) [inline .circleci/config.yml]: This is cool.. does it show stuff in a better UI?

File: .circleci/config.yml [hunk 4]
```diff
@@ -287,9 +327,10 @@ jobs:
           keys:
             - dep-cache-{{ checksum "Gopkg.toml" }}-{{ checksum "WORKSPACE" }}
```
Comment (rshriram) [inline .circleci/config.yml]: remove workspace? Not sure if this should be pinned with Gopkg.lock..

File: tools/deb/Makefile [hunk 5]
```diff
@@ -0,0 +1,2 @@
+
+deb:
```
Comment (rshriram) [inline tools/deb/Makefile]: empty?

File: tools/deb/istio-iptables.sh [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+IFS=,
+if [ -n "${IP_RANGES_INCLUDE:-}" ]; then
+    for cidr in ${IP_RANGES_INCLUDE}; do
+        iptables -t nat -A ISTIO_OUTPUT -d ${cidr} -j ISTIO_REDIRECT
+    done
+    iptables -t nat -A ISTIO_OUTPUT -j RETURN
+else
+    iptables -t nat -A ISTIO_OUTPUT -j ISTIO_REDIRECT
+fi
+
```
Comment (rshriram) [inline tools/deb/istio-iptables.sh]: is this officially replacing prepare_proxy.sh ? is this the script thats going to run in every pod?

File: .circleci/Makefile (hunk 2 above)
Comment (costinm) [reply .circleci/Makefile]: I think CircleCI will detect the change and pull ( it is a label - like latest ). 
It is only a problem in k8s, where we need to use the sha as a workaround.

If it fails - I'll do that.

File: .circleci/config.yml (hunk 3 above)
Comment (costinm) [reply .circleci/config.yml]: Example:
https://circleci.com/gh/costinm/istio/534

If a test fails it shows the failed test, and I think it has some graphs.

File: .circleci/config.yml (hunk 4 above)
Comment (costinm) [reply .circleci/config.yml]: Ongoing discussion, we'll address it in separate PR.

File: tools/deb/Makefile (hunk 5 above)
Comment (costinm) [reply tools/deb/Makefile]: Fixed (removed, I put the deb in the top makefile )

File: tools/deb/istio-iptables.sh (hunk 6 above)
Comment (costinm) [reply tools/deb/istio-iptables.sh]: This is moved from the proxy repo - it is used for raw VMs.

Eventually need to replace or merge with prepare_proxy - main difference is the default capture policy, on raw vm we don't intercecept all, just the cidrs.

File: .circleci/Makefile (hunk 2 above)
Comment (rshriram) [reply .circleci/Makefile]: circleci caches docker images. So if we update the same image, some nodes are going to have latest while others will have old.

**Pull request merged**